src/python/pants/engine/fs.py

```diff
@@ -197,12 +192,6 @@ def __init__(self, digests: Iterable[Digest]) -> None:
         """
         self.digests = tuple(digests)
 
-    def __post_init__(self) -> None:
-        non_digests = [v for v in self.digests if not isinstance(v, Digest)]  # type: ignore[unreachable]
-        if non_digests:
-            formatted_non_digests = "\n".join(f"* {v}" for v in non_digests)
-            raise ValueError(f"Not all arguments are digests:\n\n{formatted_non_digests}")
```

Comment:
No need for this validation anymore. CPython has a nice error message, so this unnecessarily slows down performance for not much value.